src/sota_tools/ostree_http_repo.cc

```diff
@@ -41,22 +43,45 @@ OSTreeRef OSTreeHttpRepo::GetRef(const std::string &refname) const { return OSTr
 OSTreeObject::ptr OSTreeHttpRepo::GetObject(const uint8_t sha256[32]) const { return GetObject(OSTreeHash(sha256)); }
 
 bool OSTreeHttpRepo::Get(const boost::filesystem::path &path) const {
+  bool ret = false;
+  CURLcode err = CURLE_OK;
   CURL *easy_handle = curl_easy_init();
```

Comment:
Is it worth checking if this returns `NULL`? I would expect that `curl_easy_perform` would break, but not sure about the `curl_easy_setopt` and `InjectIntoCurl` calls yet.

Reply:
Yes, any of these can fail in theory...

Reply:
I mean, if something causes an exception, that's sort of okay, in that we'd hear about the problem immediately. But the potential for silent failures concerns me more.